python/src/aiconfig/Config.py

```diff
@@ -274,6 +265,77 @@ async def run(
         await self.callback_manager.run_callbacks(event)
         return response
 
+    async def run_batch(
+        self,
+        prompt_name: str,
+        parameters_list: list[dict[str, Any]],
+        options: Optional[InferenceOptions] = None,
+        **kwargs,
+    ) -> list[Tuple["ExecuteResult", JSONObject | Any]]:
```

Comment:
Just return the corresponding params for that run in the Tuple as well.
So Tuple["ExecuteResult", JSONObject | Any, Dict]